src/index.ts

```diff
@@ -7,25 +7,29 @@ import {
   SetStateAction
 } from 'react';
 
+interface RequestInitStreaming extends RequestInit {
```

Comment:
Right, I think this is actually a better way so we don't need the `stream` option. Will play around with it to see what other indicators can be used to infer if streaming or not. And yes for json, we can just check the content-type.

Reply:
I was searching for some but can't really find them. I would say that structured formats with a clear start and end like JSON, XML should not be streamed, even with `Transfer-Encoding: chunked` as a partial file is invalid. Formats like text, YAML, binary could be streamed. But I guess even then the client has to know when the last chunk has arrived and the response is complete.

Reply:
Agreed, streaming is dictated by the server. But even then, there's actually no reliable way to detect it atm as `Transfer-Encoding` header isn't even available on cors request. See https://developers.google.com/web/updates/2015/03/introduction-to-fetch#response_types. 

If somehow you have access to the server, you can expose the `Transfer-Encoding` header via https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Access-Control-Expose-Headers but that's still unreliable since you're talking to different service, and don't even have access to it to be able to expose some headers.

So I think for one-off responses like json, xml, etc. we can infer those by checking the `content-type` header then calling `.json()` or `.text()` accordingly but that's about it. For everything else we return the the body (ReadableStream) and let the caller decide. Thoughts?

Reply:
It seems strange if a server allows for CORS calls, set the `Transfer-Encoding: chunked` header but then doesn't set the `Access-Control-Expose-Headers` for the client to use. But then strange things happen 😞 	

I would suggest using streaming makes sense if the client can read the `Transfer-Encoding: chunked` header and the `Content-Type` makes sense for streaming. If not use the `.json()` or `.text()` depending on the `Content-Type`.

Reply:
So just to be clear, we can't use both `Transfer-Encoding` and `Content-Type` to decide whether to stream or not. First, `Transfer-Encoding` isn't exposed to us and second, `Content-Type` has nothing to do with streaming or not since it could be `text/plain` but still streaming so we can't rely on that either.

But we could do:
1. pass option `{ streaming: true }` - return aggregated chunks as ArrayBuffer
2. no option - return `rsp.body` promise and let caller decide to call `.json()` or `.text()` or however they want to handle the data